lib/buffer.js

```diff
@@ -177,10 +173,12 @@ Buffer.from = function(value, encodingOrOffset, length) {
     return fromArrayBuffer(value, encodingOrOffset, length);
 
   if (value == null)
-    throw new TypeError(kFromErrorMsg);
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'first argument',
+      ['string', 'buffer', 'arrayBuffer', 'array', 'array-like object'], value);
 
   if (typeof value === 'number')
-    throw new TypeError('"value" argument must not be a number');
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'value', 'not number',
```

Comment:
I extend the `invalidArgType(name, expected, actual)` in [internal/errors.js](https://github.com/nodejs/node/blob/master/lib/internal/errors.js) to make it adapt to the 'not' cases.
The case here will end up being:
```
TypeError [ERR_INVALID_ARG_TYPE]: The "value" argument must not be of type number. Received type number
```